### ISSUE TITLE
panic: utf8 name validation requested but model.NameValidationScheme is not set to UTF8 in a test file

### ISSUE THREAD

**Issue by machine424** [prometheus/prometheus#15530]
### What did you do?

This is probably  a known/temporary limitation.

Adding a new test in https://github.com/prometheus/prometheus/pull/15529

got

```
2024/12/03 17:19:06 http: TLS handshake error from 127.0.0.1:52972: remote error: tls: bad certificate
--- FAIL: TestTargetsFromGroupWithLabelKeepDrop (0.00s)
    --- FAIL: TestTargetsFromGroupWithLabelKeepDrop/no_relabeling (0.00s)
panic: utf8 name validation requested but model.NameValidationScheme is not set to UTF8 [recovered]
	panic: utf8 name validation requested but model.NameValidationScheme is not set to UTF8 [recovered]
	panic: utf8 name validation requested but model.NameValidationScheme is not set to UTF8
```

(had no problem running the test locally)

### What did you expect to see?

no panic, maybe the global var should be overridden in an `init` or sth.
Not sure is this is the way to go.
how to be sure it's set for all tests files?

### What did you see instead? Under which circumstances?

Had to set `metric_name_validation_scheme: legacy` to get around the panic

**Comment by beorn7** [comment]
Maybe this will resolve itself once the UTF-8 validation is the default setting over in prometheus/common.

**Comment by ywwg** [comment]
This happens when there is a mismatch between the prometheus validation setting and the underlying common library validation setting. Make sure to set `model.NameValidationScheme = model.UTF8Validation`

**Comment by machine424** [comment]
> This happens when there is a mismatch between the prometheus validation setting and the underlying common library validation setting. Make sure to set `model.NameValidationScheme = model.UTF8Validation`

Yep, I saw we were doing this in many test files' `init()` but I wasn't sure that's the best way to go (I fell back on legacy).
I was wondering if we should make sure it's the case for all tests or sth, as someone adding a test to file without the `init()` shouldn't have to deal with this. (better UX for them I mean)
